### PR TITLE
Update default astrometry catalogs used for a posteriori processing

### PR DESCRIPTION
**Pull request by dulude**
### Associated tickets and issues
- Ticket [HLA-486](https://jira.stsci.edu/browse/HLA-486)
- Issue #931 

### High-level summary
- Updated codebase to use the new early-release Gaia DR3 (GAIAedr3) as either the default option or the first option of several to try.
- Corrected some PEP8 style issues in astrometric_utils.py.

### CODE REVIEW

**Comment by stsci-hack** [comment]
I can not find how the catalog specifications in the default_parameters JSON files get passed into the 'align_to_gaia' method of the Product classes in product.py as called from hapsequencer.   Please review the 'run_align_to_gaia' function in 'hapsequencer.py' (starting with Line 674) to make sure the 'catalog_list' entries from the JSON config files get passed as the 'catalog' parameter of the 'align_to_gaia' method.

**Comment by stsci-hack** [comment]
Photutils must be limited to <1.0.2 for this PR.  See #985 on change to 'setup.py' to get this to work.

**Comment by dulude** [comment]
> Photutils must be limited to <1.1.0 for this PR. See #985 on change to 'setup.py' to get this to work.

Resolved by commit [cd9a419](https://github.com/spacetelescope/drizzlepac/pull/986/commits/cd9a4194e88b1b5bcfe673eaed5768bcfe68ee0e)

**Comment by dulude** [comment]
> I can not find how the catalog specifications in the default_parameters JSON files get passed into the 'align_to_gaia' method of the Product classes in product.py as called from hapsequencer. Please review the 'run_align_to_gaia' function in 'hapsequencer.py' (starting with Line 674) to make sure the 'catalog_list' entries from the JSON config files get passed as the 'catalog' parameter of the 'align_to_gaia' method.

Issue resolved by commits [d65cdf1](https://github.com/spacetelescope/drizzlepac/pull/986/commits/d65cdf136855a0b236ce1def29bee4b7d54cdf76) and [b4d7624](https://github.com/spacetelescope/drizzlepac/pull/986/commits/b4d76249b23d7aab89602702d53e98b81e128bd7)

File: drizzlepac/hapmultisequencer.py [hunk 1]
```diff
@@ -354,7 +354,10 @@ def run_align_to_gaia(total_obj_list, log_level=logutil.logging.INFO, diagnostic
         #  - migrate updated WCS solutions to exp_obj instances, if necessary (probably not?)
         #  - re-run tot_obj.generate_metawcs() method to recompute total object meta_wcs based on updated
         #    input exposure's WCSs
-        align_table, filt_exposures = gaia_obj.align_to_gaia(output=diagnostic_mode, fit_label='SVM')
+        catalog_name = gaia_obj.configobj_pars.pars['alignment'].pars_multidict['all']['run_align']['catalog_list'][0]  # For now, just pass in 'catalog_name = GAIAedr3'
+        align_table, filt_exposures = gaia_obj.align_to_gaia(catalog_name=catalog_name,
+                                                             output=diagnostic_mode,
+                                                             fit_label='SVM')
```
Comment (stsci-hack) [inline drizzlepac/hapmultisequencer.py:360]: Silly mistake on my part that shows up now... 'fit_label' should actually be 'MVM' at this stage in the MVM code.

File: drizzlepac/haputils/astrometric_utils.py [hunk 2]
```diff
@@ -577,15 +572,13 @@ def build_auto_kernel(imgarr, whtarr, fwhm=3.0, threshold=None, source_box=7,
     kern_img[:, -edge:] = 0.0
     kernel_psf = False
 
-    peaks = find_peaks(kern_img, threshold=threshold * 5,
-                       box_size=isolation_size)
+    peaks = photutils.detection.find_peaks(kern_img, threshold=threshold * 5, box_size=isolation_size)
```
Comment (stsci-hack) [inline drizzlepac/haputils/astrometric_utils.py:575]: With the change in import statements for the photutils functions, this line only needs 'peaks = find_peaks('.

File: drizzlepac/haputils/astrometric_utils.py [hunk 3]
```diff
@@ -577,15 +572,13 @@ def build_auto_kernel(imgarr, whtarr, fwhm=3.0, threshold=None, source_box=7,
...
     if peaks is None or (peaks is not None and len(peaks) == 0):
         tmean = threshold.mean() if isinstance(threshold, np.ndarray) else threshold
         if tmean > kern_img.mean():
             kern_stats = sigma_clipped_stats(kern_img)
             threshold = kern_stats[2]
-        peaks = find_peaks(kern_img, threshold=threshold,
-                           box_size=isolation_size)
+        peaks = photutils.detection.find_peaks(kern_img, threshold=threshold, box_size=isolation_size)
```
Comment (stsci-hack) [inline drizzlepac/haputils/astrometric_utils.py:581]: Again, 'photutils.detection' can be removed from this line due to the change in imports at the top of the module.

**Review by stsci-hack — CHANGES_REQUESTED** [review]
A couple of changes just to clean up the calls to photutils functions.  Everything else looks good.

File: drizzlepac/hapmultisequencer.py (hunk 1 above)
Comment (dulude) [reply drizzlepac/hapmultisequencer.py]: Issue resolved by commit [8cb06c4](https://github.com/spacetelescope/drizzlepac/pull/986/commits/8cb06c43d8752b5100cfb194faffc898b657ccbe)

File: drizzlepac/haputils/astrometric_utils.py (hunk 2 above)
Comment (dulude) [reply drizzlepac/haputils/astrometric_utils.py]: issue resolved by commit [8892775](https://github.com/spacetelescope/drizzlepac/pull/986/commits/8892775dfdbf046c425976b4d1d831517eae30bf)

File: drizzlepac/haputils/astrometric_utils.py (hunk 3 above)
Comment (dulude) [reply drizzlepac/haputils/astrometric_utils.py]: issue resolved by commit [8892775](https://github.com/spacetelescope/drizzlepac/pull/986/commits/8892775dfdbf046c425976b4d1d831517eae30bf)

**Review by stsci-hack — APPROVED** [review]
This looks good.

**Pull request merged**